vpc/allocate/ip_pool_manager.go

```diff
@@ -59,7 +59,8 @@ func (mgr *IPPoolManager) assignMoreIPs(ctx *context.VPCContext, batchSize int)
 	}
 
 	originalIPCount := len(mgr.networkInterface.IPv4Addresses)
-	for i := 0; i < 10; i++ {
+	now := time.Now()
+	for time.Since(now) < ipRefreshTimeout {
```

Comment:
In the comment on line 69, aren't we not retrying in that case?

Reply:
Er, that means we retry to "lease" or "allocate" an IP from from the IPs assigned to this ENI, whereas this is to "assign" IPs. I can change the comment if it helps?